src/streaming/StreamProcessor.js

```diff
@@ -660,7 +660,7 @@ function StreamProcessor(config) {
     function _prepareForFastQualitySwitch(representationInfo) {
         // if we switch up in quality and need to replace existing parts in the buffer we need to adjust the buffer target
         const time = playbackController.getTime();
-        let safeBufferLevel = 1.5;
+        let safeBufferLevel = 1.5 * !isNaN(representationInfo.fragmentDuration) ? representationInfo.fragmentDuration : 1;
```

Comment:
This expression requires parentheses:

`let safeBufferLevel = 1.5 * (!isNaN(representationInfo.fragmentDuration) ? representationInfo.fragmentDuration : 1);` 

Otherwise the factor 1.5 was not applied in my tests